explicit-bzero/explicit-bzero.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Portable Snippets - https://github.com/nemequ/portable-snippets
+ * Created by arnavyc <arnavyc@outlook.com>
+ *
+ *   To the extent possible under law, the authors have waived all
+ *   copyright and related or neighboring rights to this code.  For
+ *   details, see the Creative Commons Zero 1.0 Universal license at
+ *   https://creativecommons.org/publicdomain/zero/1.0/
+ *
+ */
+
+#ifndef __STDC_WANT_LIB_EXT1__
+/* memset_s() function in C11 Annex K (bound-checking interface) */
+#define __STDC_WANT_LIB_EXT1__ 1
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "explicit-bzero.h"
```

Comment:
I'd prefer to not have a separate C file, and just move the entire implementation into the header.  If you move the implementation into the header than all you have to do to integrate this into your code is #include the header, whereas right now you also need to add the C file into your build system.

A separate C file is great when you have lots of code which would bloat the executable if you included a copy everywhere, or if you need to maintain global state, but in this case the implementation is so simple that I think a header-only implementation would be a much better way to go.

Reply:
I don't have any problem in doing so, however I cannot figure out a way to call `SecureZeroMemory` without including `windows.h`. Moreover, I am not using Windows so I can't do much about it.

Reply:
Done (see commit 81fc55f)